ext/reflection/php_reflection.c

```diff
@@ -3778,7 +3781,42 @@ ZEND_METHOD(reflection_class, getStaticProperties)
 	}
 
 	array_init(return_value);
-	add_class_vars(ce, 1, return_value);
+
+	ZEND_HASH_FOREACH_STR_KEY_PTR(&ce->properties_info, key, prop_info) {
+		if (((prop_info->flags & ZEND_ACC_PROTECTED) &&
+		     !zend_check_protected(prop_info->ce, ce)) ||
+		    ((prop_info->flags & ZEND_ACC_PRIVATE) &&
+		     prop_info->ce != ce)) {
+			continue;
+		}
+		prop = NULL;
+		if ((prop_info->flags & ZEND_ACC_STATIC) != 0) {
```

Comment:
Doing an early-continue here would be more elegant than setting prop to NULL and checking it below, imho.

Reply:
ACK. Done.